site/public/js/user-profile.js

```diff
@@ -121,6 +128,41 @@ function updateUserProfilePhoto () {
   return false;
 }
 
+function updateUserSecondaryEmail () {
+    const second_email = $('#user-secondary-email-change');
+    const second_email_notify = $('#user-secondary-email-notify-change');
+    if (second_email.data('current-second-email') === second_email.val() && second_email_notify.get(0).checked === (second_email_notify.data('current-second-email-notify') === 1)) {
+        displayErrorMessage('No changes detected to secondary email');
+    }
+    else {
+        let data = new FormData();
+        data.append('csrf_token', $("#user-secondary-email-csrf").val());
+        data.append('secondary_email', second_email.val());
+        data.append('secondary_email_notify', second_email_notify.get(0).checked);
+        let url = buildUrl(['user_profile', 'change_secondary_email']);
+        $.ajax({
+            url,
+            type: "POST",
+            data,
```

Comment:
I think on the success you can just refresh the page and let the twig repopulate the new fields instead of manually editing them, if we update the form in the future we would need to remember to update this logic as well. On error you can display the error message.

Also using the fetch API would clean up some of the code
```js
let response = await fetch(url , {
    method : 'POST',
    headers: {
      'Accept': 'application/json',
      'Content-Type': 'application/json'
    },
   body : { csrf_token : window.csrfToken, secondary_email .... }
});

response.json().then((data) => {
 ...
}).catch((err) => { });
```

Reply:
One issue I see with this is if you refresh the page you won't be prompted with a success message but rather just a reload. You would see the fields updated but I think the success message is nice to have. Thoughts?

In terms of using the fetch API. I see that it could clean up some of the code but everywhere else in this file also uses Ajax. I would prefer to keep Ajax here to keep it consistent overall in this file.

Reply:
You can use the `$this->core->addSuccessMessage("...");` function which will add a green success banner when the page loads, its the PHP equivalent to the JS function you used.

Reply:
Personally I feel the UI feels cleaner with the Ajax request rather than a full reload from a form request on a pop up. I think keeping it consistent with how preferred name would be good. Why do you think switching to the twig render is better?